crates/ursa-gateway/src/indexer/mod.rs

```diff
@@ -1 +1,56 @@
 pub mod model;
+
+use crate::indexer::model::IndexerResponse;
+use libp2p::multiaddr::Protocol;
+use rand::Rng;
+use std::net::SocketAddrV4;
+use tracing::{error, warn};
+
+// Randomly chooses a provider and returns its addresses.
+pub fn get_provider(indexer_response: IndexerResponse) -> Option<Vec<SocketAddrV4>> {
+    // We expect that MultihashResults will have at most 1 element.
```

Comment:
@theBeardA iirc you can have multiple

Reply:
@scott-dn

Reply:
I think Scott said this would only happen if we batched content

Reply:
@b0xtch`findCid` only returns one, `findBatch` can have multiple.
> https://github.com/ipni/storetheindex/blob/main/server/finder/http/handler.go#L57-L83

Reply:
let me clarify, multihash element can have at most one element, that's correct.

Reply:
all goods @theBeardA 👍